mapper-processor/src/test/java/com/datastax/oss/driver/internal/mapper/processor/entity/EntityNamingStrategyTest.java

```diff
@@ -0,0 +1,131 @@
+ */
+package com.datastax.oss.driver.internal.mapper.processor.entity;
+
+import com.datastax.oss.driver.api.mapper.annotations.Entity;
+import com.datastax.oss.driver.api.mapper.annotations.NamingStrategy;
+import com.datastax.oss.driver.api.mapper.entity.naming.NameConverter;
+import com.datastax.oss.driver.api.mapper.entity.naming.NamingConvention;
+import com.datastax.oss.driver.internal.mapper.processor.MapperProcessorTest;
+import com.squareup.javapoet.AnnotationSpec;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import javax.lang.model.element.Modifier;
+import org.junit.Test;
+
+public class EntityNamingStrategyTest extends MapperProcessorTest {
+
+  private static final ClassName CUSTOM_CONVERTER_CLASS_NAME =
+      ClassName.get("test", "CustomConverter");
+
+  private static final TypeSpec CUSTOM_CONVERTER_CLASS =
+      TypeSpec.classBuilder(CUSTOM_CONVERTER_CLASS_NAME)
+          .addSuperinterface(NameConverter.class)
+          .addMethod(
+              MethodSpec.methodBuilder("toCassandraName")
+                  .addModifiers(Modifier.PUBLIC)
+                  .returns(String.class)
+                  .addParameter(String.class, "javaName")
+                  .addStatement("return null;") // doesn't matter, converter won't be invoked
+                  .build())
+          .build();
+
+  // Common code for the entity class, to avoid repeating it in every test
+  private TypeSpec.Builder entityTemplate() {
+    return TypeSpec.classBuilder("TestEntity")
+        .addModifiers(Modifier.PUBLIC)
+        .addAnnotation(Entity.class)
+        .addField(TypeName.INT, "i", Modifier.PRIVATE)
+        .addMethod(
+            MethodSpec.methodBuilder("setI")
+                .addParameter(TypeName.INT, "i")
+                .addModifiers(Modifier.PUBLIC)
+                .build())
+        .addMethod(
+            MethodSpec.methodBuilder("getI")
+                .returns(TypeName.INT)
+                .addModifiers(Modifier.PUBLIC)
+                .addStatement("return 0")
+                .build());
+  }
+
+  @Test
+  public void should_fail_if_both_convention_and_converter_specified() {
+    should_fail_with_expected_error(
+        "Invalid annotation configuration: "
+            + "NamingStrategy must have either a 'convention' or 'customConverterClass' argument, "
+            + "but not both",
+        "test",
+        CUSTOM_CONVERTER_CLASS,
+        entityTemplate()
+            .addAnnotation(
+                AnnotationSpec.builder(NamingStrategy.class)
+                    .addMember("convention", "$T.CASE_INSENSITIVE", NamingConvention.class)
+                    .addMember("customConverterClass", "$T.class", CUSTOM_CONVERTER_CLASS_NAME)
+                    .build())
+            .build());
+  }
+
+  @Test
+  public void should_fail_if_neither_convention_nor_converter_specified() {
```

Comment:
I don't understand why in this case we should expect an error. Shouldn't we fallback to default naming strategy if nothing was specified?

Reply:
We fall back to the default when there is no annotation. If the annotation is present, it must have one or the other.